src/pages/spawn/spawnVolume/spawnVolumeTable/spawnVolumeTableActions/mountButton/MountVolumeModal.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from "react";
+import { useMutation, useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import { Variant } from "@leafygreen-ui/button";
+import { Disclaimer } from "@leafygreen-ui/typography";
+import { Select } from "antd";
+import { Modal } from "components/Modal";
+import { ModalContent, Section, WideButton } from "components/Spawn";
+import { InputLabel } from "components/styles";
+import { pollInterval } from "constants/index";
+import { useBannerDispatchContext } from "context/banners";
+import {
+  AttachVolumeToHostMutation,
+  AttachVolumeToHostMutationVariables,
+  MyHostsQuery,
+  MyHostsQueryVariables,
+} from "gql/generated/types";
+import { ATTACH_VOLUME } from "gql/mutations/attach-volume";
+import { GET_MY_HOSTS } from "gql/queries";
+import { useNetworkStatus } from "hooks";
+import { MyVolume } from "types/spawn";
+
+const { Option } = Select;
+
+interface Props {
+  visible: boolean;
+  onCancel: () => void;
+  volume: MyVolume;
+}
+interface HostOption {
+  key: string;
+  displayName: string;
+}
+
+export const MountVolumeModal: React.FC<Props> = ({
+  visible,
+  onCancel,
+  volume,
+}) => {
+  const dispatchBanner = useBannerDispatchContext();
+  const [hostOptions, setHostOptions] = useState<HostOption[]>([]); // dropdown option
+  const [selectedHostId, setSelectedHostId] = useState(""); // selected dropdown item
+  const { data, loading, startPolling, stopPolling } = useQuery<
+    MyHostsQuery,
+    MyHostsQueryVariables
+  >(GET_MY_HOSTS, {
+    pollInterval,
+    onError: (e) => {
+      dispatchBanner.errorBanner(
+        `There was an error loading hosts: ${e.message}`
+      );
+    },
+  });
+  useNetworkStatus(startPolling, stopPolling);
+  const targetAvailabilityZone = volume.availabilityZone;
+
+  // set host dropdown options
+  useEffect(() => {
+    if (data?.myHosts) {
+      const opts = data.myHosts
+        // Filter hosts that do not have the same availability zone as the volume.
+        .filter(
+          ({ availabilityZone }) => availabilityZone === targetAvailabilityZone
+        )
+        // Map host to a displayName and ID for the dropdown <Option />
+        .map((host) => ({
+          key: host.id,
+          displayName: host.displayName ? host.displayName : host.id,
+        }))
+        // Sort the dropdown items by display name.
+        .sort((a, b) => a.displayName.localeCompare(b.displayName));
```

Comment:
Im not sure if localeCompare is necessary any reason that you chose to use it that im not seeing? 

You can sort an array of strings using

`.sort((a,b) => a.displayName>b.displayName ? 1 : -1 )`

Reply:
I prefer localeCompare to using a ternary because it's the built in string comparison function and the return value (negative, 0, positive) is perfectly aligned to use with the `.sort()` callback